### PR TITLE
Add link health check script

### PR DESCRIPTION
**Pull request by simeg**
Here's the script I mentioned in #202.

As the description of the script says, it only exits with status code 1 if a website responds with `5xx`. All other status codes are considered as OK or warnings. You could say it's optimistically configured, but this could of course be changed.

**Action points:**
- [ ] Review and merge this PR
- [ ] Add Travis configuration on this branch
- [ ] [Configure Travis to run this branch daily](https://docs.travis-ci.com/user/cron-jobs/) @sb2nov (only owners can do this)
- [ ] [Add Travis status badge to README to show](https://docs.travis-ci.com/user/status-images/) @sb2nov (only owners can get the link)

`.travis.yml` file that should be added to **this branch only** after this PR is merged. 
```yaml
language: generic
addons:
  apt:
    packages:
      - parallel
      - grep # Untested. Used with `ggrep` command in script, might need change
script: /bin/bash ./health_check_links.sh
```

### CODE REVIEW

File: scripts/health_check_links.sh [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+echo "🔎 Finding markdown files.."
+readonly MARKDOWN_FILES_STR=$(find "$(PWD)"/.. -type f -name "*.md")
+readonly MARKDOWN_FILES_ARR=(${MARKDOWN_FILES_STR// / })
+echo "Found [" ${#MARKDOWN_FILES_ARR[@]} "] files"
+
+echo "🔬 Parsing URLs.."
+URL_ARR=()
+for FILE in "${MARKDOWN_FILES_ARR[@]}"; do
+  # Parse URL
+  URLS_STR=$(cat $FILE | ggrep -o -E 'https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)')
```
Comment (sb2nov) [inline scripts/health_check_links.sh]: http instead of https as there might be some of those?

Comment (hugovk) [reply scripts/health_check_links.sh]: The `s?` in `https?` means zero or one s. So it finds http or https.

Comment (sb2nov) [reply scripts/health_check_links.sh]: Ah yes, should have read it more carefully, thanks.

**Comment by Kyslik** [comment]
Neat! :)

**Comment by hugovk** [comment]
> * [x] Configure Travis to run this branch daily @sb2nov (only owners can do this)

I can do this and have done (with "Do not run if there has been a build in the last 24h"):

image

**Comment by hugovk** [comment]
@simeg I deleted the feature branch `health-check`. Or so we still need it? If so, please can you push it again? Thanks!

**Comment by simeg** [comment]
@hugovk Yes, I was planning on using this branch for health checking - It's restored!

**Comment by hugovk** [comment]
Thanks for restoring and enabling cron! (I'm going to enable a monthly `master` cron to ensure it is tested regularly.)

---

Re:

> * [ ] Add Travis status badge to README to show @sb2nov (only owners can get the link)

I think everyone can grab Travis badges. Go to https://travis-ci.org/sb2nov/mac-setup, click the badge at the top and choose the branch.

---

Will this require us to merge `master` into `health-check` from time to time, being careful not to overwrite or break the different `.travis.yml`?

If so, can we do better to automate it? For example, can the `health-check` build checkout the `master` branch (or just clone into another folder) and then run on that latest and greatest?

(And if so, `health-check` won't need any source but `.travis.yml`, the script and maybe a few other bits and pieces. The actual guides could be removed to keep it clean and avoid confusion.)

**Comment by simeg** [comment]
@hugovk That would be an awesome improvement, to just clone the master branch and use that as the source of data - let's do that! That would indeed make it more self-contained, less error prone (merging often causes problems) and just a cleaner solution.

I got the script running now on travis, and I'll add the badge to the master README. Feel free to make any changes and I'll be glad to do any code review.

**Pull request merged**